eng/validation/validation-functions.ps1

```diff
@@ -0,0 +1,56 @@
+set-strictmode -version 2.0
+$ErrorActionPreference = 'Stop'
+
+# Get a temporary directory for a test root. Use the agent work folder if running under azdo, use the temp path if not.
+$testRootBase = if ($env:AGENT_WORKFOLDER) { $env:AGENT_WORKFOLDER } else { $([System.IO.Path]::GetTempPath()) }
+$testRoot = Join-Path -Path $testRootBase -ChildPath $([System.IO.Path]::GetRandomFileName())
+New-Item -Path $testRoot -ItemType Directory | Out-Null
+
+function Get-LatestBuildSha()
+{
+    ## Verified that this API gets completed builds, not in progress builds
+    $headers = Get-AzDOHeaders
+    $uri = "https://dev.azure.com/${global:azdoOrg}/${global:azdoProject}/_apis/build/latest/${global:buildDefinitionId}?branchName=${global:subscribedBranchName}&api-version=5.1-preview.1"
```

Comment:
Anything that is $global: here should probably end up as parameters to the function. If the parent script doesn't set these variables, they're going to be empty when calling these functions.